From 48ccc62b3948ae963293c59694a3b6a2c2928bd1 Mon Sep 17 00:00:00 2001
From: debbuild <debbuild>
Date: Fri, 3 Aug 2018 11:32:30 +0200
Subject: [PATCH 4/7] Fix compiler warnings in common.w.

---
 common.w | 15 +++------------
 1 file changed, 3 insertions(+), 12 deletions(-)

diff --git a/common.w b/common.w
index d8aacfb..efdf891 100644
--- a/common.w
+++ b/common.w
@@ -1180,7 +1180,7 @@ concatenated to print the final error message.
 fatal(s,t)
   char *s,*t;
 {
-  if (*s) printf(s);
+  if (*s) printf("%s",s);
   err_print(t);
   history=fatal_message; exit(wrap_up());
 }
@@ -1403,16 +1403,7 @@ Several macros make other kinds of output convenient.
 @d C_printf(c,a) fprintf(C_file,c,a)
 @d C_putc(c) putc(c,C_file) /* isn't \CEE/ wonderfully consistent? */
 
-@ We predeclare several standard system functions here instead of including
-their system header files, because the names of the header files are not as
-standard as the names of the functions. (For example, some \CEE/ environments
-have \.{<string.h>} where others have \.{<strings.h>}.)
-
-@<Predecl...@>=
-extern int strlen(); /* length of string */
-extern int strcmp(); /* compare strings lexicographically */
-extern char* strcpy(); /* copy one string to another */
-extern int strncmp(); /* compare up to $n$ string characters */
-extern char* strncpy(); /* copy up to $n$ string characters */
+@ @<Include files@>=
+#include <string.h>
 
 @** Index.
-- 
2.18.0

